app/common/history.ts

```diff
@@ -0,0 +1,91 @@
+export interface Change<T> {
+    updated: T;
+    original: T;
+}
+
+function reverse<T>(change: Change<T>): Change<T> {
+    return { updated: change.original, original: change.updated };
+}
+
+export interface UpdateStyleAction {
+    type: 'update-style';
+    targets: Array<{ webviewId: string; selector: string }>;
+    style: string;
+    change: Change<string>;
+}
+
+export type Action = UpdateStyleAction;
+
+function undoAction(action: Action): Action {
+    switch (action.type) {
+        case 'update-style':
+            return {
+                type: 'update-style',
+                targets: action.targets,
+                style: action.style,
+                change: reverse(action.change),
+            };
+    }
+}
+
+interface Intransaction {
+    type: 'in-transaction';
```

Comment:
Nit: I prefer pulling raw strings into constants or enums whenever possible. Mitigates the risk of a typo introducing difficult to trace bugs

Reply:
I think Typescript guards against this, but will do

Reply:
implemented in https://github.com/onlook-dev/studio/pull/54/commits/8e0cc0446bbe1b21bf25f24dc73ee793b4000da1